### PR TITLE
Feature/clean rbf -> develop

### PR DESCRIPTION
**Pull request by RialJohnson**
- Added RBF network 
- Classification and regression
- Rusty wrote most RBF code
- I did some cleanup

### CODE REVIEW

**Review by Rusty89 — APPROVED** [review]
I approve, looks like a lot of formatting that I messed up, sorry for doing that to ya.  Don't merge without a couple more approvals though, I'm biased.

**Comment by RialJohnson** [comment]
> I approve, looks like a lot of formatting that I messed up, sorry for doing that to ya. Don't merge without a couple more approvals though, I'm biased.

No problem. Once we get both networks merged together I plan to do some more structural object-oriented changes to try and reduce some code re-use.

File: Project 3/NeuralNets/src/RBFNetwork.java
Comment (GiantFrog) [inline Project 3/NeuralNets/src/RBFNetwork.java:34]: This was my little way of informing all the layers of their previous and next layers, even when you don't know how many hidden layers there will be. Lines 20-23 do the exact same thing, though, making this part redundant. I'd get rid of either this bit or the other bit, depending on if it's at all possible you'd ever want more hidden layers.

File: Project 3/NeuralNets/src/RBFNetwork.java
Comment (GiantFrog) [inline Project 3/NeuralNets/src/RBFNetwork.java:232]:
```suggestion
            for (int maxIterations = 200; maxIterations > 0; maxIterations--) {
```

File: Project 3/NeuralNets/src/RBFNetwork.java
Comment (GiantFrog) [inline Project 3/NeuralNets/src/RBFNetwork.java:288]:
```suggestion
            for (int maxIterations = 200; maxIterations > 0; maxIterations--) {
```

**Review by GiantFrog — APPROVED** [review]

File: Project 3/NeuralNets/src/RBFNetwork.java
Comment (RialJohnson) [reply Project 3/NeuralNets/src/RBFNetwork.java:34]: Alright, will remove during cleanup since RBF will only ever have one hidden layer by definition, thanks.

File: Project 3/NeuralNets/src/RBFNetwork.java
Comment (RialJohnson) [reply Project 3/NeuralNets/src/RBFNetwork.java:232]: What's a for loop? Will fix this lol

**Pull request merged**
